### ISSUE TITLE
Localization Quirk: No AM/PM for Dansk

### ISSUE THREAD

**Issue by Docteh** [yuzu-emu/yuzu#8289]
Screen snippet from discord, not exactly repeatable on my end.
![image](https://user-images.githubusercontent.com/190571/166104476-9e8bb966-ccc5-4fd4-9e5d-8cb6e42a3819.png)

English U.S. Windows, Dansk Yuzu
![image](https://user-images.githubusercontent.com/190571/166104518-25fd3005-8f39-4884-8e6c-ccf32d4d613b.png)

But if I install Dansk, reboot windows, set the keyboard back to English, the AM/PM is now invisible.
![image](https://user-images.githubusercontent.com/190571/166104558-6f5eb9da-8d6b-47c0-87ab-0d98f265058d.png)

Windows Dansk, English Yuzu
![image](https://user-images.githubusercontent.com/190571/166105095-289170c6-6da4-4830-8e7b-3047f22799d3.png)

https://github.com/yuzu-emu/yuzu/blob/b50e1596cc2bf871478605c64674cfe544ffd43d/src/yuzu/configuration/configure_system.ui#L469-L480

GitHub's search doesn't find. 
https://github.com/yuzu-emu/yuzu/blob/b50e1596cc2bf871478605c64674cfe544ffd43d/src/yuzu/configuration/configure_system.cpp#L74-L75

https://doc.qt.io/qt-5/qdatetime.html

> Note: If localized month and day names are desired, please switch to using [QLocale::system](https://doc.qt.io/qt-5/qlocale.html#system)().[toString](https://doc.qt.io/qt-5/qdatetime.html#toString)() as [QDateTime](https://doc.qt.io/qt-5/qdatetime.html) methods shall change to use English (C locale) names at Qt 6.

I'm going to add more details to this later, just sniffing out the problem as Localization quirks interest me.

**Comment by Docteh** [comment]
```diff
diff --git a/src/yuzu/configuration/configure_system.cpp b/src/yuzu/configuration/configure_system.cpp
index 19aa589f9..99946426c 100644
--- a/src/yuzu/configuration/configure_system.cpp
+++ b/src/yuzu/configuration/configure_system.cpp
@@ -71,6 +71,8 @@ void ConfigureSystem::SetConfiguration() {
     ui->rng_seed_edit->setText(rng_seed);

     ui->custom_rtc_checkbox->setChecked(Settings::values.custom_rtc.has_value());
+    //Hack?
+    ui->custom_rtc_edit->setLocale(QLocale::English);
     ui->custom_rtc_edit->setEnabled(Settings::values.custom_rtc.has_value());
     ui->custom_rtc_edit->setDateTime(QDateTime::fromSecsSinceEpoch(rtc_time));


```

![image](https://user-images.githubusercontent.com/190571/166106885-3f210f92-a5f5-46d1-9a95-79c875d79814.png)

Whoops I might not have responded to this over on discord. 
![image](https://user-images.githubusercontent.com/190571/166107027-acb68eda-f160-4e99-acb4-d20b930ba125.png)

@german77 The oddity here is that Yuzu explicitly asks for AM/PM to be shown in the date/time widget with 12 hour values. With my current locale settings the hour scrolls from 12PM to 11PM (0 through 23 hours) but there is no AM or PM shown

so it scrolls 12 1 2 3 4 5 6 7 8 9 10 11 12 1 2 3 4 5 6 7 8 9 10 11 

Solution A: Change the UI to be 24 hour for everyone
This will probably JustWork(TM) so I'm ignoring it for now. I can PR it if asked. 

Solution B: Change the UI to be 24 hour only when necessary
`ui->custom_rtc_edit->setDisplayFormat(QStringLiteral("d MMM yyyy hh:mm:ss"));`
But I'd need to figure out the "when necessary" part.

Solution C: Change the UI to be properly English 12 hour mode
`ui->custom_rtc_edit->setLocale(QLocale::English);` in that diff above

Solution D: Remove the format specified in the UI file and see what happens in English and Dansk and maybe sucker someone else to test a locale

**Comment by german77** [comment]
I think I know what's going on. For some reason the date format can be edited by the translation files. Most people don't know that these affect the date so they put wrong values.
![image](https://user-images.githubusercontent.com/5944268/166119151-4fcdca72-60c4-4c06-8b71-049319dec6dd.png)

Solution B would be the easiest way to solve this issue. It's not a matter of what the system clock has. It's the translation file altering the values.
`ui->custom_rtc_edit->setDisplayFormat(QStringLiteral("d MMM yyyy h:mm:ss AP")`

**Comment by german77** [comment]
Feel free to make a PR or let me know and I can do it

**Comment by lioncash** [comment]
Generally you want to allow modification of the date string because some languages and locales may have a different ordering in how dates are rendered, The best course of action is to fix the relevant translation file and add an explanatory disambiguation text to the translation, similar to how `tr` allows you to do:

```cpp
tr("some text to translate", "text that clarifies what the text means")
```

**Comment by Docteh** [comment]
es id nl tr_TR have translations that look like they would break this field
```
es.ts:        <source>d MMM yyyy h:mm:ss AP</source>
es.ts-        <translation>d MMM aaaa h:mm:ss AP</translation>
--
id.ts:        <source>d MMM yyyy h:mm:ss AP</source>
id.ts-        <translation>AP h BBB tttt j:mm:dd</translation>
--
nl.ts:        <source>d MMM yyyy h:mm:ss AP</source>
nl.ts-        <translation>d MMM jjjj u:mm:ss AP</translation>
--
tr_TR.ts:        <source>d MMM yyyy h:mm:ss AP</source>
tr_TR.ts-        <translation>g AAA yyyy s:dd:snsn AP</translation>

```

However the Dansk translation just passes the string through
```
$ grep "<source>d MMM" -A 1 da.ts
        <source>d MMM yyyy h:mm:ss AP</source>
        <translation>d MMM yyyy h:mm:ss AP</translation>
```

Test of Solution D (remove format entirely)
[   4.115240] Frontend <Critical> yuzu\configuration\configure_system.cpp:SetConfiguration:78: rtc display format: dd-MM-yyyy HH:mm

![image](https://user-images.githubusercontent.com/190571/166126367-d012bc55-65c6-4f46-a34e-6fa70fb35e4c.png)


@lioncash Would leaving it up to the System locale be good enough? 

If we want the Yuzu UI language to influence the Date Time widget, we'd have to give the Widget a reasonable QLocale that matches the given language.

**Comment by lioncash** [comment]
Yeah, leaving it up to system locale is fine

**Comment by lioncash** [comment]
Addressed by #8291
